scylla/src/frame/value_tests.rs

```diff
@@ -718,3 +719,53 @@ async fn test_cqlvalue_udt() {
 
     assert_eq!(received_udt_cql_value, &udt_cql_value);
 }
+
+#[tokio::test]
+async fn test_cqlvalue_duration() {
+    let uri = env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+    let session: Session = SessionBuilder::new().known_node(uri).build().await.unwrap();
+    session
+        .query(
+            "CREATE KEYSPACE IF NOT EXISTS ks WITH REPLICATION = \
+            {'class' : 'SimpleStrategy', 'replication_factor' : 1}",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    session
+        .query(
+            "CREATE TABLE IF NOT EXISTS ks.cqlvalue_duration_test (k int, my duration, primary key (k))",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    let duration_cql_value = CqlValue::Duration(CqlDuration {
+        months: 6,
+        days: 9,
+        nanoseconds: 21372137,
+    });
+
+    session
+        .query(
+            "INSERT INTO ks.cqlvalue_duration_test (k, my) VALUES (5, ?)",
+            (&duration_cql_value,),
+        )
+        .await
+        .unwrap();
+
+    let rows = session
+        .query("SELECT my FROM ks.cqlvalue_duration_test", &[])
+        .await
+        .unwrap()
+        .rows
+        .unwrap();
+
+    assert_eq!(rows.len(), 1);
+    assert_eq!(rows[0].columns.len(), 1);
+
+    let received_duration_cql_value = rows[0].columns[0].as_ref().unwrap();
+
+    assert_eq!(received_duration_cql_value, &duration_cql_value);
```

Comment:
In this test, please also check that a duration inserted with a CQL query in which the duration is encoded in the CQL string is then deserialized properly by the driver.